tests/callbacks/test_image_visualizer.py

```diff
@@ -44,3 +47,64 @@ def test_image_visualizer(interval: str):
     assert isinstance(image_visualizer.interval, Time)
     assert num_train_tables == (num_train_steps - 1) // image_visualizer.interval.value + 1
     assert num_eval_tables == 1
+
+
+@pytest.mark.skipif(not _WANDB_INSTALLED, reason='Wandb is optional')
+def test_wandb_and_image_visualizer():
+    os.environ['WANDB_MODE'] = 'dryrun'
+
+    import wandb
+
+    image_interval = 2
+    image_visualizer = ImageVisualizer(interval=f'{image_interval}ba')
+    wandb_logger = WandBLogger()
```

Comment:
Would it be cleaner to provide `mode='offline'` to the `wandb.init`? (e.g. https://docs.wandb.ai/ref/python/init)

or alternatively, maybe `os.environ[WANDB_MODE]='offline'` should be a fixture that runs across the entire codebase so we are never dependant on wandb uptime?

Reply:
Also, anyway to set the file dir to a `tmpdir` fixture?

Reply:
yes, I'll use a tmpdir and pass offline to init. I think we can leave it up to individual tests whether or not to use offline mode. currently this would be the only test running that actually uses the `WandBLogger` as far as i can tell.

Reply:
@hanlint It turned out there was already a fixture that was setting mode to `disabled` for tests not market remote, so I just changed that to `offline` instead. Does that make sense? Or is that going to mess with other tests?

Reply:
Ah, that makes sense!